api/oneinch/order.go

```diff
@@ -115,17 +73,16 @@ func (client *Client) PlaceOrder(params orderbook.CreateOrderParams) error {
 	}
 	beforeRequest()
 	defer afterRequest()
-	request, err := oneInchClient.NewRequest("POST", fmt.Sprintf("/orderbook/v3.0/%d", params.ChainId), body)
+
+	_, _, err = oneInchClient.Orderbook.CreateOrder(context.Background(), params)
 	if err != nil {
-		return err
-	}
-	var response orderbook.CreateOrderResponse
-	if _, err := oneInchClient.Do(context.Background(), request, &response); err != nil {
-		return err
-	}
-	if !response.Success {
-		return errors.New("an unknown error occured. your limit order did not get saved")
+		if strings.Contains(err.Error(), "1inch router does not have approval for this token") {
```

Comment:
I don't like how you are comparing an error with a string. Can we introduce a strongly typed error for this?

Reply:
More context regarding this demo PR [here](https://github.com/svanas/ladder/pull/3#discussion_r1501850744).

Here is a follow up PR with strongly a typed error: https://github.com/1inch/1inch-sdk/pull/27